src/main/cpp/src/row_conversion.cu

```diff
@@ -2259,8 +2259,15 @@ std::unique_ptr<table> convert_from_rows(lists_column_view const &input,
     for (int i = 0; i < static_cast<int>(schema.size()); ++i) {
       if (schema[i].id() == type_id::STRING) {
         // stuff real string column
-        auto const null_count = string_row_offset_columns[string_idx]->null_count();
         auto string_data = string_row_offset_columns[string_idx].release()->release();
+        auto const null_count = [&] {
+          // Null-count not set previously. Calculate, on the fly.
+          auto const &null_mask = *string_data.null_mask;
+          return null_mask.data() ?
+                     cudf::null_count(static_cast<bitmask_type const *>(null_mask.data()), 0,
```

Comment:
```suggestion
                     cudf::null_count(reinterpret_cast<bitmask_type const *>(null_mask.data()), 0,
```

Reply:
`null_mask.data()` is a `void*`. There is a standards-defined conversion from `void*` to `T*`. Is there a reason here to use `reinterpret_cast` when `static_cast` suffices?

Reply:
Oh, interesting: 
1. Per [some readings](http://jdebp.info/FGA/reinterpret-cast-and-pointer-to-void.html) of the Standard, using `reinterpret_cast` here would constitute bad code. TIL. I don't have a copy of the standards to confirm.
2. Stroustrup seems to have [changed](https://groups.google.com/g/comp.lang.c++.moderated/c/rcgB5ZRooIE/m/rvv5UQhyDCUJ) from using `reinterpret_cast` to `static_cast`, for `void* -> T*`, in the "Special Edition" of TC++PL.

Looks like `static_cast` is the right call here.

Reply:
Keeping this as `static_cast`. I'm open to discussing this if required.